src/main/java/nextstep/ladder/domain/Line.java

```diff
@@ -0,0 +1,57 @@
+package nextstep.ladder.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.IntFunction;
+import java.util.stream.Collectors;
+
+public class Line {
+    private final List<Boolean> points;
+
+    public Line(final int width, final IntFunction<Boolean> lineBuilderStrategy) {
+        this.points = new ArrayList<>(width);
+
+        for (int idx = 0; idx < width; idx++) {
+            addPoint(idx, lineBuilderStrategy.apply(idx));
+        }
+    }
+
+    List<Boolean> getPoints() {
+        return points;
+    }
+
+    private void addPoint(final int idx, boolean curPoint) {
+        if (idx > 0) {
+            boolean prePoint = this.points.get(idx - 1);
+
+            curPoint = adjustContinuousTrue(prePoint, curPoint);
+        }
+
+        this.points.add(curPoint);
+    }
+
+    private boolean adjustContinuousTrue(final boolean prePoint, boolean curPoint) {
+        if (prePoint && curPoint) {
```

Comment:
이 조건은 `prePoint` 가 `true` 인 경우 항상 `false` 를 반환하게 되므로, 아래와 같이 변경해도 괜찮을 것 같네요!
```suggestion
        if (prePoint) {
```

Reply:
생각해보니 맞는 말씀이신 것 같습니다. 
좋은 의견 감사합니다!👍👍